src/Compilers/CSharp/Portable/FlowAnalysis/DataFlowPassBase.cs

```diff
@@ -40,10 +41,21 @@ internal abstract partial class DataFlowPassBase<TLocalState> : AbstractFlowPass
             Symbol member,
             BoundNode node,
             EmptyStructTypeCache emptyStructs,
-            bool trackUnassignments)
+            bool trackUnassignments,
+            ImmutableDictionary<VariableIdentifier, int> variableSlot = null,
+            ImmutableArray<VariableIdentifier> variableBySlot = default)
             : base(compilation, member, node, trackUnassignments: trackUnassignments)
         {
             _emptyStructTypeCache = emptyStructs;
+            if (variableSlot != null)
+            {
+                foreach (var pair in variableSlot)
+                {
+                    _variableSlot.Add(pair.Key, pair.Value);
+                }
+                this.variableBySlot = variableBySlot.ToArray();
+                nextVariableSlot = variableBySlot.Length;
+            }
```

Comment:
It seems dangerous to assume that this is the number of the next available slot unless you assert all slots in variableSlot are less than this.